src/app/Main/Header/Export/Header-ExportContainer.component.tsx

```diff
@@ -98,15 +123,60 @@ export function HeaderExportContainerComponent(): JSX.Element {
                 statData,
                 statKeys,
             },
-        });
-    };
+        };
+
+        if ((!raw && stat) || (raw && !stat)) {
+            const dataString: string = dataOrStatisticToCsvString({
+                setting: newSettings,
+                filter: filter.selectedFilter,
+                allFilterLabel: exportLabels.allFilterLabel,
+                mainFilterLabels: exportLabels.mainFilterLabels,
+                mainFilterAttributes: filter.mainFilter,
+            });
+            setCsvDataString(dataString)
+            /* exportSubmitButton = (
+                <div>
+                    {exportButtonLabel}
+                    <CSVDownload
+                        data={dataString}
+                        filename={ZNFilename}
+                        target="_blank"
+                    />
+                </div>
+            );
+            clearData() */
+        }
+
+        if (raw && stat) {
+            dataAndStatisticToZipFile({
+                setting: newSettings,
+                ZNFilename,
+                filter: filter.selectedFilter,
+                allFilterLabel: exportLabels.allFilterLabel,
+                mainFilterLabels: exportLabels.mainFilterLabels,
+                mainFilterAttributes: filter.mainFilter,
+                subFileNames,
+            });
+        }
+        setSetting(newSettings);
+        setLoading(false);
+    };    
+
+    // https://stackoverflow.com/questions/64817556/loading-data-asynchronously-and-downloading-csv-data-with-a-click-is-one-step-be
 
     useEffect(() => {
         if (isExport) {
             fetchAndChooseData(setting.raw, setting.stat);

```

Comment:
i would set is export=false only here, then the logic becomes clearer. Something like:
```
setIsLoading(true);
await fetchAndChooseData(...)
setIsLoading(false);
setIsExport(false);
```